### PR TITLE
:book: Versioning and Branching Strategy

### PR DESCRIPTION
**Pull request by DirectXMan12**
This is mostly a proposal at this point.  I've started to outline the branch strategy we've loosely discussed in the past, with some motivations and discussion of alternatives.

### CODE REVIEW

**Comment by DirectXMan12** [comment]
@droot @pwittrock PTAL and let me know what you think

File: VERSIONING.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+accrued.  Since we don't intend to have a ton of these, the maintainers
+will evaluate when to do a major release as it comes up.
+
+### Exact Steps
+
+***TODO***
+
+### Breaking Changes
+
+Try to avoid breaking changes.  They make life difficult for users, who
```
Comment (pwittrock) [inline VERSIONING.md]: Can we provide a `sed` operation to automatically fix any breaking changes with defaults that maintain the old behavior or provide some new default?

Comment (DirectXMan12) [reply VERSIONING.md]: Probably want more intelligent than sed, but it should be possible to do so

File: VERSIONING.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+- We follow [Semantic Versioning (semver)](https://semver.org)
+- Use releases with your dependency management to ensure that you get
+  compatible code
+- The master branch contains all the latest code, some of which may break
+  compatibility (so "normal" `go get` is not reccomended)
+
+### Contributors
+
+- All code PR must be labeled with :bug: (patch fixes), :sparkles:
+  (backwards-compatible features), or :warning: (breaking changes)
```
Comment (droot) [inline VERSIONING.md]: Github PR template can help us here.

**Pull request merged**